src/main/scala/sbtghpackages/GitHubPackagesPlugin.scala

```diff
@@ -43,14 +43,14 @@ object GitHubPackagesPlugin extends AutoPlugin {
   val authenticationSettings = Seq(
     githubTokenSource := TokenSource.Environment("GITHUB_TOKEN"),
 
-    credentials += {
+    credentials ++= {
       val src = githubTokenSource.value
       inferredGitHubCredentials("_", src) match {   // user is ignored by GitHub, so just use "_"
         case Some(creds) =>
-          creds
+          Seq(creds)
 
         case None =>
-          sys.error(s"unable to locate a valid GitHub token from $src")
+          Seq.empty
```

Comment:
So the problem I have with this is the fact that *resolution* from GitHub Packages also requires a token. You can't just download a package unauthenticated, meaning that credentials are necessary at all times regardless of whether or not you're publishing. Honestly, this is a thing that GitHub needs to fix.

Reply:
Won’t the authentication issue arise at the time of resolution or publication though ie aren’t we doubling up on the checking here?

Reply:
It would arise at the time of resolution. Basically this is trying to give users a better error message. They won't be able to *use* the plugin to either publish or resolve without a token, so just trying to give a message which is semantic rather than some opaque "could not resolve artifact from one of these resolvers" error.

Reply:
I’m not sure that this is the correct place to improve on resolution messages, but your call. Happy for you to close this PR though.